### PR TITLE
fix pkg/controller/resourcequota static check

### PR DESCRIPTION
**Pull request by panpan0000**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug

/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:fixes pkg/controller/resourcequota static check errors.

**Which issue(s) this PR fixes**:
Part of https://github.com/kubernetes/kubernetes/issues/92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/resourcequota/resource_quota_controller_test.go [hunk 1]
```diff
@@ -1137,8 +1137,6 @@ func expectSyncNotBlocked(fakeDiscoveryClient *fakeServerResources, workerLock *
 
 	workerLockAcquired := make(chan struct{})
 	go func() {
-		workerLock.Lock()
```
Comment (panpan0000) [inline pkg/controller/resourcequota/resource_quota_controller_test.go:1140]: the reason why I removed them : static-check complains for a blank critical zone here.

Comment (panpan0000) [reply pkg/controller/resourcequota/resource_quota_controller_test.go:1140]: maybe is it better for something dummy code between Lock and Unlock...

Comment (lavalamp) [reply pkg/controller/resourcequota/resource_quota_controller_test.go:1140]: This looks very intentional, I hope a test fails when you remove it like this :)

Comment (panpan0000) [reply pkg/controller/resourcequota/resource_quota_controller_test.go:1140]: hmm.... I got you . maybe the static-check pass but the unit-test failed. will dig into it. thank you

**Comment by BenTheElder** [comment]
/unassign
someone else will need to review this

**Pull request closed without merging**